pkgs/servers/monitoring/fusion-inventory/default.nix

```diff
@@ -1,18 +1,56 @@
-{ stdenv, fetchurl, buildPerlPackage, perlPackages
+{ stdenv, fetchurl, buildPerlPackage, perlPackages, gnused, nix, dmidecode, pciutils, usbutils, iproute, nettools
+, fetchFromGitHub, fetchpatch
 }:
 
 buildPerlPackage rec {
-  version = "2.3.18";
+  version = "2.3.21";
   name = "FusionInventory-Agent-${version}";
-  src = fetchurl {
-    url = "mirror://cpan/authors/id/G/GR/GROUSSE/${name}.tar.gz";
-    sha256 = "543d96fa61b8f2a2bc599fe9f694f19d1f2094dc5506bc514d00b8a445bc5401";
+  src = fetchFromGitHub {
+    owner = "fusioninventory";
+    repo = "fusioninventory-agent";
+    rev = "2.3.21";
+    sha256 = "034clffcn0agx85macjgml4lyhvvck7idn94pqd2c77pk6crvw2y";
   };
 
-  patches = [ ./remove_software_test.patch ];
+  patches = [
+    ./remove_software_test.patch
+    # support for os-release file
+    (fetchpatch {
+      url = https://github.com/fusioninventory/fusioninventory-agent/pull/396.diff;
+      sha256 = "0iv9jf2v531xz7a3dfqh83cvadhwlk7685y2l3ypy2kg0ybi96pm";
+    })
+    # support for Nix software inventory
+    (fetchpatch {
+      url = https://github.com/fusioninventory/fusioninventory-agent/pull/397.diff;
+      sha256 = "17bjv7zyhcfjjq7yg5p5089jymn2fp2kd5kaldv4y0wyhy7ij3ad";
+    })
+    ];
 
   postPatch = ''
+    # The os-release patch file contains a rename, but patch does not apply this correctly
+    mv lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/LSB.pm \
+       lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/OSRelease.pm
```

Comment:
Try `fetchurl` instead of `fetchpatch`. See #32084.

Reply:
Thanks for the hint, ``fetchurl`` indeed works better.